src/windows/ride.c

```diff
@@ -2360,15 +2405,46 @@ static void window_ride_main_invalidate(rct_window *w)
 
 	window_ride_anchor_border_widgets(w);
 
+	const int offset = gCheatsAllowArbitraryRideTypeChanges ? 15 : 0;
 	// Anchor main page specific widgets
 	window_ride_main_widgets[WIDX_VIEWPORT].right = w->width - 26;
-	window_ride_main_widgets[WIDX_VIEWPORT].bottom = w->height - 14;
+	window_ride_main_widgets[WIDX_VIEWPORT].bottom = w->height - (14 + offset);
 	window_ride_main_widgets[WIDX_STATUS].right = w->width - 26;
-	window_ride_main_widgets[WIDX_STATUS].top = w->height - 13;
-	window_ride_main_widgets[WIDX_STATUS].bottom = w->height - 3;
+	window_ride_main_widgets[WIDX_STATUS].top = w->height - (13 + offset);
+	window_ride_main_widgets[WIDX_STATUS].bottom = w->height - (3 + offset);
 	window_ride_main_widgets[WIDX_VIEW].right = w->width - 60;
 	window_ride_main_widgets[WIDX_VIEW_DROPDOWN].right = w->width - 61;
 	window_ride_main_widgets[WIDX_VIEW_DROPDOWN].left = w->width - 71;
+	window_ride_main_widgets[WIDX_RIDE_TYPE].right = w->width - 87;
+	window_ride_main_widgets[WIDX_RIDE_TYPE].top = w->height - 17;
+	window_ride_main_widgets[WIDX_RIDE_TYPE].bottom = w->height - 4;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_INCREASE].right = w->width - 88;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_INCREASE].left = w->width - 98;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_INCREASE].top = w->height - 16;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_INCREASE].bottom = w->height - 11;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_DECREASE].right = w->width - 88;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_DECREASE].left = w->width - 98;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_DECREASE].top = w->height - 10;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_DECREASE].bottom = w->height - 5;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_APPLY].left = w->width - 83;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_APPLY].right = w->width - 26;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_APPLY].top = w->height - 17;
+	window_ride_main_widgets[WIDX_RIDE_TYPE_APPLY].bottom = w->height - 4;
+
+	widget_invalidate(w, WIDX_VIEWPORT);
+	widget_invalidate(w, WIDX_VIEW);
```

Comment:
This is a large performance hit, you should only invalidate things when necessary.